### ISSUE TITLE
RakNet Security layer is not supported

### ISSUE THREAD

**Issue by RaphiMC** [Sandertv/go-raknet#28]
Trying to connect to a server which uses the RakNet security flag (In OpenConnectionReply1) causes a timeout. While I am not 100% sure that the timeout is not caused by some other error in go-raknet other RakNet libraries failed because of the servers usage of the security flag (See https://github.com/CloudburstMC/Network/issues/37). Connecting to the server with Bedrock Edition (or https://github.com/extremeheat/fb-raknet) works fine since both use the original RakNet.

Reproduceable example code:
```go
package main

import (
	"github.com/sandertv/go-raknet"
	"time"
)

func main() {
	conn, err := raknet.DialTimeout("play.timecrack.net:19132", time.Duration(5)*time.Second)
	if err != nil {
		panic(err.Error())
	}
	defer func(conn *raknet.Conn) {
		err := conn.Close()
		if err != nil {
			panic(err.Error())
		}
	}(conn)

	println("Connected")
}
```

**Comment by dktapps** [comment]
FWIW, the client decodes OpenConnectioRequest1 differently depending on whether it was compiled with security support or not. If the server enables security but the client doesn't support it, it won't read the key bytes from the packet, and will interpret those bytes as the MTU size. Basically the server needs to be psychic and know whether the client supports encryption in order to structure the packet properly. RakNet Quality Code ™

**Comment by Sandertv** [comment]
Fixed in [1beb5ce](https://github.com/Sandertv/go-raknet/commit/1beb5ceb09b78d4a8d44284dd9e6466cb72e2d6d).
